core/src/main/java/hudson/model/Fingerprint.java

```diff
@@ -1324,7 +1328,17 @@ public Api getApi() {
         }
 
         Fingerprint loaded = FingerprintStorage.get().load(id);
```

Comment:
Would be better to cache the storage implementations as local variables (current storage and then the Filesystem one). Apart from performance benefits, it reduces the chance of the logic exploding in unlikely case when the Storage gets reconfigured in the middle of the `load()` operation. Multi-threading 🤷

Reply:
I agree, thanks for pointing it out :) Done!